packages/react-components/src/components/chart/utils/getInfo.ts

```diff
@@ -0,0 +1,68 @@
+import { DurationViewport, Viewport } from '@iot-app-kit/core/src';
+import { v4 as uuid } from 'uuid';
+import { DEFAULT_MARGIN, tooltipLineColor, tooltipLineWidth } from '../eChartsConstants';
+import { ElementEvent } from 'echarts';
+import { SizeConfig } from '../types';
+export const isDurationViewport = (viewport: Viewport): viewport is DurationViewport =>
+  (viewport as DurationViewport).duration !== undefined;
+
+// TODO: test this once echarts live mode is supported
+export const viewportToSeconds = (viewport?: Viewport) => {
+  if (viewport && isDurationViewport(viewport)) {
+    switch (viewport?.duration) {
+      case '1m':
+        return { widthInMs: 60 * 1000, initial: Date.now() - 60 * 1000 };
+      case '5m':
+        return { widthInMs: 60 * 1000 * 5, initial: Date.now() - 60 * 1000 * 5 };
+      case '10m':
+        return { widthInMs: 60 * 1000 * 10, initial: Date.now() - 60 * 1000 * 10 };
+      case '30m':
+        return { widthInMs: 60 * 1000 * 30, initial: Date.now() - 60 * 1000 * 30 };
+      default:
+        return { widthInMs: 60 * 1000 * 10, initial: Date.now() - 60 * 1000 * 10 };
+    }
+  } else {
+    const start = new Date(viewport?.start ?? 0).getTime();
+    const end = new Date(viewport?.end ?? 0).getTime();
+    return {
+      widthInMs: end - start,
+      initial: start,
+    };
+  }
+};
+
+export const calculateTimeStamp = (xInPixel: number, widthInPixel: number, viewport?: Viewport) => {
+  const { widthInMs, initial } = viewportToSeconds(viewport);
+  // TODO: this results in a decimal , needs to decide precision
+  const delta = (widthInMs * (xInPixel - DEFAULT_MARGIN)) / (widthInPixel - 2 * DEFAULT_MARGIN);
+  return delta + initial;
+};
+
+export const addNewGraphic = (e: ElementEvent, size: SizeConfig, viewport: Viewport | undefined) => ({
+  id: `tooltip-${uuid()}`,
+  action: '$merge',
+  type: 'group',
+  timestampInMs: calculateTimeStamp(e.offsetX, size?.width, viewport),
+  children: [
+    {
+      type: 'line',
+      z: 101,
```

Comment:
we should leave some comments about why this number specifically or make it a variable in the other echarts constants. would be nice to have 1 place for all the layering info so that it's easy to debug